rust/crd/src/authentication.rs

```diff
@@ -0,0 +1,290 @@
+    "sn".to_string()
+}
+
+fn default_email_field() -> String {
+    "mail".to_string()
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassLdapBindCredentials {
+    /// [SecretClass](https://docs.stackable.tech/secret-operator/secretclass.html) containing the LDAP bind credentials
+    pub secret_class: String,
+    /// [Scope](https://docs.stackable.tech/secret-operator/scope.html) of the [SecretClass](https://docs.stackable.tech/secret-operator/secretclass.html)
+    pub scope: Option<AuthenticationClassSecretClassScope>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassSecretClassScope {
+    #[serde(default)]
+    pub pod: bool,
+    #[serde(default)]
+    pub node: bool,
+    #[serde(default)]
+    pub services: Vec<String>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum AuthenticationClassTls {
+    /// Use TLS but don't verify certificates.
+    /// We have to use an empty struct instead of an empty Enum because of limitations of [kube-rs](https://github.com/kube-rs/kube-rs/)
+    Insecure {},
+    /// Use TLS and the ca certificates provided by the system - in this case the Docker image - to verify the server.
+    /// This can be useful when you e.g. use public AWS S3 or other public available services.
+    SystemProvided {},
+    /// Use TLS and ca certificate to verify the server
+    ServerVerification(AuthenticationClassTlsServerVerification),
+    /// Use TLS and ca certificate to verify the server and the client
+    MutualVerification(AuthenticationClassTlsMutualVerification),
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassTlsServerVerification {
+    /// Ca cert to verify the server
+    pub server_ca_cert: AuthenticationClassCaCert,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AuthenticationClassTlsMutualVerification {
+    /// [SecretClass](https://docs.stackable.tech/secret-operator/secretclass.html) which will provide ca.crt, tls.crt and tls.key
+    pub secret_class: String,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub enum AuthenticationClassCaCert {
+    /// Name of the ConfigMap containing the ca cert. Key must be "ca.crt".
+    Configmap(String),
+    /// Name of the Secret containing the ca cert. Key must be "ca.crt".
+    Secret(String),
+    /// Path to the ca cert
+    Path(String),
+    /// Name of the SecretClass which will provide the ca cert
+    SecretClass(String),
+}
+
+impl AuthenticationClass {
+    pub fn append_volumes_and_volume_mounts(
```

Comment:
Do we really want to set the paths and mount names here? Wouldn't it give operators more flexibility to return a volume from this and have the operator configure the mounts as needed?

Reply:
I think it makes sense to not remove the volume_mounts as the names of the volumes and mounts have to match. If we separate them the operator has to have knowledge how the volumes are named.
The method now looks like `pub fn get_volumes_and_volume_mounts(&self) -> (Vec<Volume>, Vec<VolumeMount>)`
Then the operator can decide to use the volumes and/or the volume_mounts.

Reply:
Whoever is mounting them can always just inspect the `Volume::name`, or overwrite the name as desired.

Reply:
@teozkr and me decided on removing constructing volumes and volumemounts entirely